### DISCUSSION TITLE
Deletion of own account

### DISCUSSION THREAD

**Discussion by fvdaak** [keycloak/keycloak#13322] [Q&A] [answered]
Hello!

We'ld like to give our users the possibility to delete their own accounts. 
By our research, we found the following documentation:
https://github.com/keycloak/keycloak-documentation/blob/main/server_admin/topics/users/allow-user-to-delete-account.adoc

...but it seems to be outdated and does not work for us. If we follow the steps as described, we do not see a delete button in the user-profile.

We are running keycloak in version 18.0.0 in community edition.

Any help would be appreciated

Thank you and regards,
Frank

**Comment by ssilvert** [comment, marked as answer]
@fvdaak This only works with the latest React-based account console.  You will need to go to Realm settings --> Themes and set the account theme to "keycloak.v2".

**Reply by fvdaak** [reply]
Ahhh - nice! That does the trick :-)

Thank you very much for your help

**Reply by ch4mpy** [reply]
@ssilvert following the [latest doc I found](https://github.com/keycloak/keycloak/blob/main/docs/documentation/server_admin/topics/users/proc-allow-user-to-delete-account.adoc) and your answer above, I have:
- set `keycloak.v2` as theme for both `Account theme` and `Admin UI theme` under `Realm settings` -> `Themes` (for all my realms, so no doubt)
- enabled `Delete Account` under `Authentication` -> `Required actions` (same, all realms)
- updated to Keycloak `22.0.1`

But, when visiting [the account page](https://oidc.c4-soft.com/auth/realms/spring-addons/account/#/personal-info), I still don't see the option to delete an account I just created.

Additionally I can't see any `account delete-account` role in the assignment popup I get in admin console when editing a user.

What should I do to enable users to delete their account by themself? This is a privacy requirement in my country and might be a show stopper for using Keycloak if not possible easily.

**Reply by ssilvert** [reply]
The `delete-account` role is available as a client role of the `account` client.  So to assign that role to a user you will need to filter by clients in the `Assign roles` modal dialog.

**Reply by ch4mpy** [reply]
@ssilvert thank you for your quick answer and directions, it works!

At first, it seemed a bit unnatural to go through a sub menu of `Realm settings` to automatically add a **client** role. But after all, there is a distinct `account` client for each realm, and more importantly, users are created for the realm, not the client, so after I realized that, I understood why I should have searched in `Realm settings`.

I'm not sure why this `delete-account` role is not assigned by default when `Delete account` toggle is enabled in `Required actions`. Aren't roles required to edit the account auto assigned as soon as we enable `Update profile`?

Last, maybe should [the documentation source](https://github.com/keycloak/keycloak/blob/main/docs/documentation/server_admin/topics/users/proc-allow-user-to-delete-account.adoc) be updated to explain that this `delete-account` role is a client role of the `account` client? And also that the feature is available only with the `keycloak.v2` theme?
